### PR TITLE
HOV-16 | Divider component

### PR DESCRIPTION
**Pull request by saurabhsutar192**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/docs/components/divider.mdx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+</div>
+```
+
+> **Note:** Vertical Divider's height will depend on container's height
+
+<div style={{ height: "100px", margin: "0 auto", width: "fit-content" }}>
+  <Divider orientation="vertical" />
+</div>
+
+##### Vertical Divider with Label
```
Comment (pruthvi2103) [inline docs/docs/components/divider.mdx:76]: can you please do an example with horizontal list and vertical divider?

should look like

```
Example 1 | Example 2 | Example 3
```

File: lib/src/components/Divider/Divider.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        });
+
+  return (
+    <div
+      className={`divider-container ${
+        orientation === "horizontal"
+          ? dividerContainerHorizontal
+          : ` ${dividerContainerVertical}`
+      }`}
+    >
```
Comment (pruthvi2103) [inline lib/src/components/Divider/Divider.tsx:48]: please add a role of `seperator` 

read here about it: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/separator_role

File: lib/src/components/Divider/Divider.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+      <div
+        className={`divider ${dividerClass} `}
+        style={assignInlineVars({
+          [dividerThemeVar.dividerColor]: color,
+          [dividerThemeVar.dividerSize]: size,
+        })}
+        {...nativeProps}
```
Comment (pruthvi2103) [inline lib/src/components/Divider/Divider.tsx:55]: nativeProps will override className and style. destructure `className` and `style` and merge with styles and classNames of the component (the className on the rightmost wins so take care of that the consumers get maximum customization

File: lib/src/components/Divider/Divider.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+      ></div>
+      {label && (
+        <span
+          style={assignInlineVars({
+            [dividerThemeVar.labelColor]: labelColor,
+            [dividerThemeVar.labelBackground]: labelBackground,
+            [dividerThemeVar.dividerSize]: size,
+          })}
+          className={`label ${labelClass} `}
```
Comment (pruthvi2103) [inline lib/src/components/Divider/Divider.tsx:64]: what is the `label` className here? do not put simple classnames like these as they may clash with consuming project. always put classNames from vanilla-extract as they are unique

File: lib/src/components/Divider/Divider.tsx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+  return (
+    <div
+      className={`divider-container ${
+        orientation === "horizontal"
+          ? dividerContainerHorizontal
+          : ` ${dividerContainerVertical}`
+      }`}
+    >
+      <div
+        className={`divider ${dividerClass} `}
```
Comment (pruthvi2103) [inline lib/src/components/Divider/Divider.tsx:50]: what is the `divider` className here? do not put simple classnames like these as they may clash with consuming project. always put classNames from vanilla-extract as they are unique

File: lib/src/components/Divider/divider.css.ts [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+  position: "relative",
+  width: "100%",
+});
+
+export const dividerContainerVertical = style({
+  position: "relative",
+  height: "100%",
+  width: "fit-content",
+});
+
```
Comment (pruthvi2103) [inline lib/src/components/Divider/divider.css.ts:23]: isnt this possible doing with recipie?

File: lib/src/components/Divider/Divider.tsx [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+          type,
+        });
+  const labelClass =
+    orientation === "horizontal"
+      ? labelHorizontal({
+          labelPosition,
+        })
+      : labelVertical({
+          labelPosition,
+        });
```
Comment (pruthvi2103) [inline lib/src/components/Divider/Divider.tsx:39]: we could use a simple recipie here instead of the logic here?

**Review by pruthvi2103 — CHANGES_REQUESTED** [review]
Some comments

File: lib/src/components/Divider/Divider.tsx (hunk 7 above)
Comment (saurabhsutar192) [reply lib/src/components/Divider/Divider.tsx:39]: I initially started with a recipe, but it ended up having too many conditions...so to improve the code readability gone with this approach.

File: lib/src/components/Divider/divider.css.ts (hunk 6 above)
Comment (saurabhsutar192) [reply lib/src/components/Divider/divider.css.ts:23]: same comment as below

File: lib/src/components/Divider/Divider.tsx (hunk 5 above)
Comment (saurabhsutar192) [reply lib/src/components/Divider/Divider.tsx:50]: in order for user to customize or override any default styles...he can simply use these classNames to customize

File: lib/src/components/Divider/Divider.tsx (hunk 4 above)
Comment (saurabhsutar192) [reply lib/src/components/Divider/Divider.tsx:64]: same comment as below

File: lib/src/components/Divider/Divider.tsx (hunk 3 above)
Comment (saurabhsutar192) [reply lib/src/components/Divider/Divider.tsx:55]: okay will do that

File: lib/src/components/Divider/Divider.tsx (hunk 2 above)
Comment (saurabhsutar192) [reply lib/src/components/Divider/Divider.tsx:48]: sure

File: docs/docs/components/divider.mdx (hunk 1 above)
Comment (saurabhsutar192) [reply docs/docs/components/divider.mdx:76]: sure

File: lib/src/components/Divider/Divider.tsx (hunk 5 above)
Comment (pruthvi2103) [reply lib/src/components/Divider/Divider.tsx:50]: not really an elegant way of making people use our classnames, will not be helpful if people want to use multiple styles or something like tailwind. rather use native classname and put it beside our classes so they can customize it

eg `className={${dividerClass} ${nativeClassName}}`

gives freedom to user to give classname and wont pollute the dom, dont want the divider class to pollute all other instances of dividers in an app

File: lib/src/components/Divider/Divider.tsx (hunk 7 above)
Comment (pruthvi2103) [reply lib/src/components/Divider/Divider.tsx:39]: what i am saying is keep 2 seperate recipies, 1 for the wrapping div and the other one you are already using, will be cleaner this way

File: lib/src/components/Divider/Divider.tsx (hunk 5 above)
Comment (saurabhsutar192) [reply lib/src/components/Divider/Divider.tsx:50]: okay

File: lib/src/components/Divider/Divider.tsx (hunk 7 above)
Comment (pruthvi2103) [reply lib/src/components/Divider/Divider.tsx:39]: based on discussion with @saurabhsutar192 recipes will make the code complex so we will keep the current code

**Review by pruthvi2103 — APPROVED** [review]
LGTM

**Pull request merged**